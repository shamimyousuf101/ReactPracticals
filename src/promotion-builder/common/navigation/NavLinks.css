.navigation ul li {
  border-radius: 20%;
  border-right: solid white;
  border-top: solid white;
  box-shadow: -10px 5px 5px silver;
  cursor: pointer;
  height: 45px;
  margin-bottom: 7px;
  width: 65px;
}

ul {
  list-style: none;
}

.link__search {
  background-color: #60aade;
}

.link__asset {
  background-color: #e85441;
}

.link__edit {
  background-color: #506bb1;
}

a {
  color: white;
  cursor: pointer;
  text-decoration: none;
}
