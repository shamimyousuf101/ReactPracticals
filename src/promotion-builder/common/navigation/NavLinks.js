import React from "react";
import PropTypes from "prop-types";
import bem from "bem-cn";
import { Link } from "react-router-dom";

import "./NavLinks.css";


const b = bem("link");

const NavLinks = ({ navId, label, clickHandler, url= '#' }) => (
  <li className={b(navId.toLowerCase())} onClick={clickHandler}>
    <Link to={url}>{label}</Link>
  </li>
);

NavLinks.propTypes = {
  navId: PropTypes.string, 
  label: PropTypes.string,
  clickHandler: PropTypes.func,
  url: PropTypes.string
};

export default NavLinks;
