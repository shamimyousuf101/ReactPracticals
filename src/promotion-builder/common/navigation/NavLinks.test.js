import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';

import NavLinks from "./NavLinks";

describe("NavLink", () => {
  
  let clickHandlerMock=jest.fn();

  it('renders correctly', () => {
    const tree = renderer
      .create(<NavLinks
        label={"mobile"}
        navId={"1"}
        key={"1"}
        clickHandler={clickHandlerMock}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('on clicking the link', () => {
    it('clickHandler is called once', () => {
      const wrapper = shallow(<NavLinks
        label={"mobile"}
        navId={"1"}
        key={"1"}
        clickHandler={clickHandlerMock}
      />);
      wrapper.find('li').simulate('click');
      expect(clickHandlerMock).toHaveBeenCalledTimes(1)
    });
  })

});
