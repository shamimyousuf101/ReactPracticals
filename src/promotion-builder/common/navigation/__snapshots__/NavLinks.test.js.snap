// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink renders correctly 1`] = `
<li
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  onClick={[MockFunction]}
>
  <a
    href="#"
  >
    mobile
  </a>
</li>
`;
